tools/docs/config.ts

```diff
@@ -51,7 +84,7 @@ function genTable(obj: [string, string][], type: string, def: any): string {
         if (Object.keys(el[1] ?? []).length === 0) {
           return;
         }
-        el[1] = `\`${JSON.stringify(el[1])}\``;
+        el[1] = `<pre lang="json">${stringify(el[1], { indent: 2 })}</pre>`;
```

Comment:
does this have syntax highlight in final docs?

Reply:
Looks like it does not :[
![image](https://user-images.githubusercontent.com/24865016/173252494-ef6fd4a0-0c2d-4c7a-b9d6-a7bbe5bec65e.png)

looks good though regardless imho when compared to:

![image](https://user-images.githubusercontent.com/24865016/173252792-f6805b5d-a74e-4e2f-a756-a9f4a886d50d.png)

also, we are currently also using html tables, so there must be some kind translation process from code generated markdown tables to the final html docs.
so, would generating the docs with html tables from the get go could be of a problem?

Reply:
I still prefer the new screenshot to the old

Reply:
Yeah, the new screenshot looks _way_ better. I can read and understand the content at once.

Syntax highlighting in the tables would be nice to have, but if you can't get it working, then we should still merge the prettier tables. 😉

Reply:
From what i gathered, html 5 doesn't support syntax highlight.
could we use something like [highlight.js](https://www.npmjs.com/package/highlight.js) perhaps?

What we saw in the markdown mocks is GH's extended markdown support for syntax highlighting

Reply:
Material for MkDocs supports syntax highlighting. I guess there might be a way to get it working in the table as well.

Here are the docs:

- https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/#highlight
- https://squidfunk.github.io/mkdocs-material/reference/code-blocks/

Reply:
When inspecting [packageRules](https://docs.renovatebot.com/configuration-options/#packagerules) for example.
The source of it is a ` ```json ...``` ` block inside the `configuration-options.md` file, while inside the generated mkdocs page we have this:
```html
<p><code>packageRules</code> is a powerful feature that lets you apply rules to individual packages or to groups of packages using regex pattern matching.</p>
<p>Here is an example if you want to group together all packages starting with <code>eslint</code> into a single branch/PR:</p>
<div class="highlight"><pre><span></span><code><span class="p">{</span><span class="w"></span>
<span class="w">  </span><span class="nt">&quot;packageRules&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="w"></span>
<span class="w">    </span><span class="p">{</span><span class="w"></span>
<span class="w">      </span><span class="nt">&quot;matchPackagePatterns&quot;</span><span class="p">:</span><span class="w"> </span><span class="p">[</span><span class="s2">&quot;^eslint&quot;</span><span class="p">],</span><span class="w"></span>
<span class="w">      </span><span class="nt">&quot;groupName&quot;</span><span class="p">:</span><span class="w"> </span><span class="s2">&quot;eslint packages&quot;</span><span class="w"></span>
<span class="w">    </span><span class="p">}</span><span class="w"></span>
<span class="w">  </span><span class="p">]</span><span class="w"></span>
<span class="p">}</span><span class="w"></span>
</code></pre></div>
```

so its autogenerated while the docs are being constructed. 
Is there a way i can generated the mkdocs myself for experimenting?
it is worth checking if having this backticked block of code inside a table will produce the highlighted syntax we want.

Reply:
yes, checkout our docs repo und change the git submodule to your branch